### PR TITLE
Support for selenium 2.52.0, java-client 3.3.0, and more

### LINKED ISSUE #232: Upgrade to Selenium 2.52.0
Selenium 2.49.1 is now available - http://docs.seleniumhq.org/download/

### LINKED ISSUE #233: Upgrade to appium java-client 3.3.0
appium java-client 3.3.0 which depends on selenium 2.48.2 is available.

### LINKED ISSUE #252: FileExtractor#getExecutableNames fails on Windows 10
This change broke the mentioned method -- https://github.com/paypal/SeLion/commit/379872cc06bdf5f4bae221c397e64abb0e4212cd#diff-4926659960fb939909378dc955c38e0b in 1.1.0-SNAPSHOT

### PR DESCRIPTION
**Pull request by mach6**
-  Fix #232 - Bump support for selenium to 2.52.0
  - Fix element deselect test that requires select to be of type
    multiple select
  - _Note_: **changes will not be backwards compatible with
    selenium version < 2.49.0** .. _this is an upstream breakage_
  - _Also Note_: Users must clear or update their existing
    `${selionHome}/config/download.properties` file
- Fix #233 - Bump support for java-client to 3.3.0
- Bump guava version to 19.0
- Fix #252, #253, #254

### CODE REVIEW

File: server/src/main/java/com/paypal/selion/grid/SeLionGridLauncher.java [hunk 1]
```diff
@@ -108,43 +196,35 @@ public void boot(String[] args) throws Exception {
...
+    }
+
+    private static void printInfoAboutRoles(CommandLineOptionHelper helper) {
+        if (helper.hasParamValue("-role")) {
+            CliUtils.printWrappedLine("", "Error: the role '" + helper.getParamValue("-role")
+                    + "' does not match a recognized server role\n");
+        } else {
+            CliUtils.printWrappedLine("",
+                    "Error: -role option needs to be followed by the value that defines role of this component in the grid\n");
         }
```
Comment (mach6) [inline server/src/main/java/com/paypal/selion/grid/SeLionGridLauncher.java:221]: Codacy Issue found: [System.out.println is used](https://www.codacy.com/app/SeLion/paypal-SeLion/file/2120612344/issues/source?bid=2928425&fileBranchId=3146721#l222)

Comment (mach6) [inline server/src/main/java/com/paypal/selion/grid/SeLionGridLauncher.java:221]: Will not fix

**Comment by mach6** [comment]
@sebady @sundaramrajendran your input is requested here. :)

**Comment by sundaramrajendran** [comment]
@mach6 Looks good to me

File: server/src/main/java/com/paypal/selion/grid/SeLionGridLauncher.java [hunk 2]
```diff
@@ -96,11 +176,19 @@ public void shutdown() throws Exception {
      */
     public void boot(String[] args) throws Exception {
         CommandLineOptionHelper helper = new CommandLineOptionHelper(args);
+        GridRole role = GridRole.find(args);
+
+        if (role == null) {
+            printInfoAboutRoles(helper);
+            return;
```
Comment (sebady) [inline server/src/main/java/com/paypal/selion/grid/SeLionGridLauncher.java:183]: Should we default to a role of NOT_GRID if -role is not specified in args?

Comment (mach6) [inline server/src/main/java/com/paypal/selion/grid/SeLionGridLauncher.java:183]: `GridRole.find()` defaults to `GridRole#NOT_GRID` if `-role` is not specified for us but can also return `null` on errors. Thus, the reason for this check and the default message.  For example, `-role astronaut` will trigger the `find` to return `null`

**Pull request closed without merging**